engine/jobmaster/dm/api.go

```diff
@@ -140,6 +131,15 @@ func (jm *JobMaster) QueryJobStatus(ctx context.Context, tasks []string) (*JobSt
 	}
 	wg.Wait()
 
+	state, err := jm.metadata.UnitStateStore().Get(ctx)
+	if err != nil && errors.Cause(err) != metadata.ErrStateNotFound {
+		return nil, err
+	}
+	unitState, existUnitState = state.(*metadata.UnitState)
+	if existUnitState {
+		jobStatus.FinishedUnitStatus = unitState.FinishedUnitStatus
```

Comment:
Seems we rely on https://github.com/pingcap/tiflow/blob/e0bce5e400144a499ae08f39f4eae7ba3f2f38e1/engine/jobmaster/dm/dm_jobmaster.go#L242 is happened before https://github.com/pingcap/tiflow/blob/e0bce5e400144a499ae08f39f4eae7ba3f2f38e1/engine/jobmaster/dm/dm_jobmaster.go#L264 ?
May left some unit tests or comment to avoid developer changed the order in future

Reply:
that's a bug to reorder those 2 calls, the finished-status might not be persisted

Reply:
I prefer we also add a comment at `onWorkerFinished`. We rely on

onWorkerFinished: unitStateStore.ReadModifyWrite -> onWorkerFinished: messageAgent.RemoveClient -> QueryJobStatus: QueryStatus -> QueryJobStatus: UnitStateStore().Get

Reply:
https://github.com/pingcap/tiflow/pull/7711/commits/27cea5a3efb819f56d323811e574cfa697ec4b61